bin/expo-setup.js

```diff
@@ -19,28 +21,20 @@ if (!isPlainObject(appJson) || !isPlainObject(appJson.expo)) {
   );
 }
 
-appJson.expo.experiments ||= {};
-// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- Set to empty object in previous line, if falsy
-appJson.expo.experiments.typedRoutes = true;
-
-appJson.expo.plugins = [
-  [
-    'expo-build-properties',
-    {
-      ios: {
-        newArchEnabled: true,
-      },
-      android: {
-        newArchEnabled: true,
-      },
-    },
-  ],
-];
-
-await writeFile(appFilePath, JSON.stringify(appJson, null, 2), 'utf8');
-console.log(
-  '✅ Enabled New Architecture and typedRoutes experiment in app.json',
-);
+const expoConfig = `import { ExpoConfig } from "expo/config";
+
+const config: ExpoConfig = ${JSON.stringify(appJson.expo, null, 2)};
+
+export default config;`.trim();
+
+await writeFile('app.config.ts', expoConfig, 'utf8');
+console.log('✅ Converted app.json to app.config.ts');
+
+await promisify(exec)('npx prettier --write app.config.ts');
+console.log('✅ Formatted app.config.ts with Prettier');
```

Comment:
Need to run the [Prettier command](https://prettier.io/docs/en/cli.html#:~:text=Use%20the%20prettier%20command%20to%20run%20Prettier%20from%20the%20command%20line.&text=To%20run%20your%20locally%20installed,This%20overwrites%20your%20files!)) with `npx`, because we are using the locally installed Prettier version and not the global version.

> To run your locally installed version of Prettier, prefix the command with npx or yarn (if you use Yarn), i.e. npx prettier --help, or yarn prettier --help.

Reply:
let's use [the Prettier API](https://prettier.io/docs/en/api.html)

Reply:
Commented in another thread. 
https://github.com/upleveled/eslint-config-upleveled/pull/462#discussion_r1926822853